crates/reth/evm/src/utils.rs

```diff
@@ -22,19 +23,29 @@ pub fn wei_to_sats(wei: U256) -> (U256, U256) {
 }
 
 /// Collects withdrawal intents from bridge-out events in the receipts.
-/// Returns a vector of `WithdrawalIntent`.
+/// Returns a vector of optional [`WithdrawalIntent`]s.
+///
+/// # Note
+///
+/// A [`Descriptor`], if invalid does not create an [`WithdrawalIntent`].
 pub fn collect_withdrawal_intents(
     receipts: impl Iterator<Item = Option<Receipt>>,
-) -> impl Iterator<Item = WithdrawalIntent> {
+) -> impl Iterator<Item = Option<WithdrawalIntent>> {
     receipts
         .flatten()
         .flat_map(|receipt| receipt.logs)
         .filter(|log| log.address == BRIDGEOUT_ADDRESS)
         .filter_map(|log| {
             WithdrawalIntentEvent::decode_log(&log, true)
-                .map(|evt| WithdrawalIntent {
-                    amt: evt.amount,
-                    dest_pk: evt.dest_pk,
+                .map(|evt| {
+                    let descriptor = Descriptor::from_bytes(&evt.destination);
+                    match descriptor {
+                        Ok(valid_descriptor) => Some(WithdrawalIntent {
+                            amt: evt.amount,
+                            destination: valid_descriptor,
+                        }),
+                        Err(_) => None,
```

Comment:
This return type change doesn't seem right.  What sense is there returning incrementally `None` here?  We know the precompile will always only emit properly-formed logs.

Also, should we avoid hardcoding the address?

Reply:
I agree. we could probably use `filter_map` here and keep the return type same. The caller doesn't gain anything significant with the `None`s imo(except the count of invalid descriptors).

Reply:
Done in 26d6b54c7cb3bff32de1476c245cb76dcffbf6a0